src/interactions/rbe.jl

```diff
@@ -0,0 +1,127 @@
+struct RBEInteractions{T} <: AbstractInteraction
+    α::T
+    p::Int
+    k_set::Vector{}
+    prob::Vector{T}
+    V::Float64
+    cutoff::T
+    k_c::T
+end
+LinearAlgebra.norm(p::Point) = sqrt(norm2(p))
+norm2(p::Point) = sum(abs2, p.coo)
+norm2(p::AbstractVector) = sum(abs2, p)
+Base.zero(p::Point{N, T}) where {N, T} = zero(Point{N, T})
+Base.zero(::Type{Point{N, T}}) where {N, T} = Point(ntuple(_->zero(T), N))
+Base.zeros(::Type{Complex{Float64}}) = Complex{Float64}(0.0, 0.0)
+Base.zeros(::Type{Complex{Float64}}, p::Int) = fill(Complex{Float64}(0.0, 0.0), p)
+
+function RBEInteractions(α::T, L_x::Float64, L_y::Float64, L_z::Float64, p::Int, s::Float64) where T
+    k_c = 2s/sqrt(α)
+    cutoff = k_c / 2
+    S = calculate_S(α, L_x, L_y, L_z)
+    V = L_x * L_y * L_z
+    k_set = generate_k_set(L_x, L_y, L_z, k_c)
+    prob = compute_probabilities(k_set, α, S)
+    return RBEInteractions{T}(α, p, k_set, prob, V, cutoff, k_c)
+end
+
+function generate_k_set(L_x::Float64, L_y::Float64, L_z::Float64, k_c::Float64)
+    m_max_x = floor(Int, k_c * L_x / (2 * π))
+    m_max_y = floor(Int, k_c * L_y / (2 * π))
+    m_max_z = floor(Int, k_c * L_z / (2 * π))
+    k_set = []
+    for m1 in -m_max_x:m_max_x
+        for m2 in -m_max_y:m_max_y
+            for m3 in -m_max_z:m_max_z
+                k_vector = [2 * π * m1 / L_x, 2 * π * m2 / L_y, 2 * π * m3 / L_z]
+                if norm(k_vector) <= k_c && norm(k_vector) != 0
+                    push!(k_set, k_vector)
+                end
+            end
+        end
+    end
+    return k_set
+end
+
+function compute_probabilities(k_set, α::Float64, S::Float64)
+    probabilities = []
+    for k in k_set
+        prob = exp(-norm(k)^2 / (4 * α)) / S
+        push!(probabilities, prob)
+    end
+    return probabilities
+end
+
+function calculate_H(α::Float64, L_x::Float64, L_y::Float64, L_z::Float64)
+    Hx = sqrt(α * L_x^2 / π) * (1 + 2 * exp(-α * L_x^2))
+    Hy = sqrt(α * L_y^2 / π) * (1 + 2 * exp(-α * L_y^2))
+    Hz = sqrt(α * L_z^2 / π) * (1 + 2 * exp(-α * L_z^2))
+    return Hx * Hy * Hz
+end
+ 
+function calculate_S(α::Float64, L_x::Float64, L_y::Float64, L_z::Float64)
```

Comment:
may be we can directly sum up all $e^{-k^2/4\alpha}$ instead of this approximation